### ISSUE TITLE
[Reading] SALMON: Self-Alignment with Instructable Reward Models

### ISSUE THREAD

**Issue by hiroaki-1-1-0** [hiroaki-1-1-0/Reading-Papers#11]
## 0. Info
- Author: Zhiqing Sun, Yikang Shen, Hongxin Zhang, Qinhong Zhou, Zhenfang Chen, David Cox, Yiming Yang, Chuang Gan
- Paper: [link](https://arxiv.org/abs/2310.05910)

## 1. In a Nutshell
- Prepare training data according to the principles without using manually labeled data, perform SFT, and then perform RLAIF withe the principles as safety alignment.

## 2. Novelty / Utility
- Use the principles not only for SFT, but also for RLAIF.

## 3. Core Method
1. Collect principle-driven synthetic preferences.
2. Train a principle-conditioned reward model.
3. PPO fine-tune the policy while randomly sampling a few principles per prompt.
4. Optional symbolic bonuses (length, multilingual)

<img width="777" alt="Image" src="https://github.com/user-attachments/assets/2af9df51-02ab-4627-8298-88e7922059a7" />

## 4. Validation
- Compare the score of the proposed method with the baseline method.

## 5. Limitations / Discussions
- Principle design/coverage is tricky.
- Principle effectiveness is task-dependent.

## 6. Further Reading
- Sun et al. (2023), “Principle-Driven Self-Alignment”: Bootstraps with principles and few exemplars.

## 7. Personal Comments
- Generate reward scores based on preference guideline, too. Filled the gap left by Constitutional AI.
